test/polygon/qidao/qidao.test.ts

```diff
@@ -0,0 +1,270 @@
+      const brwAmt = parseEther("1");
+      const setVaultId = "13571113";
+      const spells = [
+        {
+          connector: connectorName,
+          method: "createVault",
+          args: [vaults.matic.address, setVaultId],
+        },
+        {
+          connector: connectorName,
+          method: "deposit",
+          args: [tokens.matic.address, vaults.matic.address, 0, amt, setVaultId, 0, 0, 0],
+        },
+        {
+          connector: connectorName,
+          method: "borrow",
+          args: [vaults.matic.address, 0, brwAmt, setVaultId, 0, 0 , 0]
+        },
+        {
+          connector: connectorName,
+          method: "payback",
+          args: [vaults.matic.address, 0, brwAmt, setVaultId, 0, 0 , 0],
+        },
+        {
+          connector: connectorName,
+          method: "withdraw",
+          args: [tokens.matic.address, vaults.matic.address, 0, amt.mul(995).div(1000), setVaultId, 0, 0, 0],
+        },
+      ];
+
+      const tx = await dsaWallet0
+        .connect(wallet0)
+        .cast(...encodeSpells(spells), await wallet1.getAddress());
+
+      await tx.wait();
+
+      expect(await ethers.provider.getBalance(dsaWallet0.address)).to.eq(
+        parseEther("9.975")
+      );
+    });
+
+
+    it("should create a LINK vault in QiDao and deposit LINK into that vault", async function() {
+      const amt = parseEther("50");
+      const brwAmt = parseEther("10");
+      const setVaultId = "13571113";
+      const spells = [
+        {
+          connector: connectorName,
+          method: "createVault",
+          args: [vaults.link.address, setVaultId],
+        },
+        {
+          connector: connectorName,
+          method: "deposit",
+          args: [tokens.link.address, vaults.link.address, 0, amt, setVaultId, 0, 0, 0],
+        },
+        {
+          connector: connectorName,
+          method: "borrow",
+          args: [vaults.link.address, 0, brwAmt, setVaultId, 0, 0 , 0]
+        },
+        {
+          connector: connectorName,
+          method: "payback",
+          args: [vaults.link.address, 0, brwAmt, setVaultId, 0, 0 , 0],
+        },
+        {
+          connector: connectorName,
+          method: "withdraw",
+          args: [tokens.link.address, vaults.link.address, 0, amt.mul(995).div(1000), setVaultId, 0, 0, 0],
+        },
+      ];
+
+      const tx = await dsaWallet0
+        .connect(wallet0)
+        .cast(...encodeSpells(spells), await wallet1.getAddress());
+
+      await tx.wait();
+
+      expect(await ethers.provider.getBalance(dsaWallet0.address)).to.eq(
+        parseEther("9.975")
+      );
+    });
+    // it("Should borrow and payback half DAI from Aave V2", async function() {
+    //   const amt = parseEther("100"); // 100 DAI
+    //   // const setId = "83478237";
+    //   await addLiquidity("dai", dsaWallet0.address, parseEther("1"));
+    //   let spells = [
+    //     {
+    //       connector: connectorName,
+    //       method: "borrow",
```

Comment:
Why are these commented ?

Reply:
Copy and paste hold over from porting the old Aave tests. Will delete

Reply:
Okay